tools/quake3/q3map2/bsp.cpp

```diff
@@ -698,11 +698,13 @@ int BSPMain( int argc, char **argv ){
 	strClear( tempSource );
 
 	/* set standard game flags */
+	maxLMSurfaceVerts = game->maxLMSurfaceVerts;
```

Comment:
I've hit this too; the question is, if this is needed; this affects meta surfaces creation, and default value may be more efficient.
This might be considered as fine with default and tuneable by experienced users; i'v seen -fastmeta commit in xonotic, forcing this to low value for efficiency.

Reply:
Just as with keeplights. These are game default definitions. We filled them expecting them to be used by default in our games, but they are never applied.

Reply:
This is what i'v said. The question is practical effect of actually using given values.
Setting might have been added e.g. for reference, not for actual use.

Reply:
The only effect on the keeplights one is that the light entities are kept in the bsp file instead of deleted as default. AFAIK this is transparent for all games since undefined entities are simply ignored at spawn. And in any case, as long as that game has that value defined as false it's definitely unafected.

Reply:
I don't know what else to tell you. I think all the definitions in the game_<whateverZ.h should be used, otherwise I don't see the point in having the fields in the game definitions, but it's your call now to decide. I'd just say that the values that aren't going to be used should be removed from the definitions because otherwise it's very missleading.

Reply:
I totally agree, but this is conception from perfect world. In real case it would be better to evaluate actual effects of the given variable change.
I think lets experiment and merge this, mainly basing on ET setting -mv in their build anyway. Hopefully DP people will report issues, caused by this, if any.

Reply:
> This might be considered as fine with default and tuneable by experienced users; i'v seen -fastmeta commit in xonotic, forcing this to low value for efficiency.

I saw that commit too. I'm not sure if I agree that it's more efficient. 3D cards love big trisoups now. Sending more meshes could actually be much less efficient. It's the main reason why I have been revising that vertexlit models are not sent through meta. They become really slow. The impact of this is very hard to find out with lightmapped ones, tho, because they are sent through meta anyway, and meta always breaks them up into pieces.

Reply:
Yeah, fewer draw calls was always faster. I imply efficiency of map building; tearing off small meta pieces may be faster, examining/reallocating huge meta may be very slow.

Reply:
Oh, yeah. I wouldn't worry about that. Meta loves breaking them up. A little too much if you ask me =)

Reply:
Well, just tested quickly, shaderlab_terrain model, _sa 179:
![image](https://user-images.githubusercontent.com/1734117/120718431-ed194f80-c4d1-11eb-8473-69b55f94a6dd.png)

Reply:
Same with given _sa 120, `7 total meta surfaces`, six for the box and one for 20k triangles terrain.

Reply:
Really? I try so hard to avoid them being split in small chunks...

Maybe because I use blocksize?